### DISCUSSION TITLE
Moving credentials away from a strictly environmental concept

### DISCUSSION THREAD

**Discussion by rtyler** [delta-io/delta-rs#279] [General]
I have been working on [riverbank](https://github.com/delta-incubator/riverbank) which has the need for specifying multiple buckets which _may_ have multiple sets of authentication variables. At present, `DeltaTable` just assumes that the storage provider can figure out credentials on its own.

I think we need to change this to allow the caller to provide some form of a credential or credential provider, but I'm not sure how to accomplish this given the myriad of ways that credentials can be provider for AWS alone.

Coincidentally I saw this related issues this morning #278

**Comment by houqp** [comment]
I think it's a good time to promote `new` table method https://github.com/delta-io/delta-rs/blob/main/rust/src/delta.rs#L890 for more advanced usage. Relevant storage backend implementations need to be updated to support auth configuration in constructor methods.

**Comment by houqp** [comment]
While we revamp storage backend abstraction, might worth consider https://github.com/delta-io/delta-rs/issues/257 as well.
